test/Controller/DownloadControllerTest.php

```diff
@@ -51,9 +52,57 @@ public function it_downloads_a_file()
         $this->assertSame(file_get_contents($mp3), $content);
     }
 
-    private function createDownloadUri(string $fileUri, string $name) : string
+    /**
+     * @test
+     */
+    public function it_may_set_the_rel_canonical_link_header()
     {
-        $uri = 'http://localhost/download/'.base64_encode($fileUri)."/$name";
+        $client = static::createClient();
+
+        $this->mockApiResponse(
+            new Request(
+                'GET',
+                'http://www.example.com/test.mp3',
+                [
+                    'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+                    'Referer' => 'http://www.example.com/',
+                    'X-Forwarded-For' => '127.0.0.1',
+                    'X-Forwarded-Host' => 'localhost',
+                    'X-Forwarded-Port' => '80',
+                    'X-Forwarded-Proto' => 'http',
+                ]
+            ),
+            new Response(
+                200,
+                ['Content-Type' => 'audio/mp3'],
+                fopen($mp3 = __DIR__.'/../../assets/tests/blank.mp3', 'r')
+            )
+        );
+
+        $content = $this->captureContent(function () use ($client) {
+            $client->request('GET', $this->createDownloadUri('http://www.example.com/test.mp3', 'test.mp3', 'http://www.example.com/canonical'), [], [], ['HTTP_REFERER' => 'http://www.example.com/']);
+        });
+
+        $response = $client->getResponse();
+
+        $this->assertInstanceOf(StreamedResponse::class, $response);
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertArraySubset([
+            'content-type' => ['audio/mp3'],
+            'content-disposition' => ['attachment; filename="test.mp3"'],
```

Comment:
I appreciate the canonical links can apply to anything, but could we use a pdf in the example here?

Reply:
Have amended the test with a pdf download. The code change allows us to set a canonicalUri for any download.